### PR TITLE
Update kubebuilder to v3

### LINKED ISSUE #36: use kubebuilder v3.0.0
## What

We still use v2.3.1.

## How

TBD

## Checklist

- [ ] Finish implementation of the issue
- [ ] Test all functions
- [ ] Have enough logs to trace activities
- [ ] Notify developers of necessary actions

### PR DESCRIPTION
**Pull request by bells17**
resolve #36

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,73 +22,110 @@ export KUBEBUILDER_ASSETS
...
-	go build -o bin/manager main.go
+##@ Build
+
+.PHONY: build
+build: generate fmt vet ## Build manager binary.
+	go build -o $(BINDIR)/manager main.go
 
 .PHONY: run
-run: generate manifests
+run: manifests generate fmt vet ## Run a controller from your host.
```
Comment (toshipp) [inline Makefile:87]: I think fmt and vet are not needed here.

**Comment by toshipp** [comment]
Looks good, please go ahead.

File: Makefile (hunk 1 above)
Comment (bells17) [reply Makefile]: fixed it.
https://github.com/topolvm/pvc-autoresizer/pull/41/commits/83985304dfcc931888d4865f415d36deb396012c

File: config/rbac/role.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   creationTimestamp: null
-  name: controller
+  name: manager-role
```
Comment (toshipp) [inline config/rbac/role.yaml:7]: If you change the name of the ClusterRole, you also change ClusterRoleBinding.
BTW, why did you change it? By kubebuilder?

**Review by satoru-takeuchi — DISMISSED** [review]

**Comment by satoru-takeuchi** [comment]
Oops, I approved during review by mistake.

File: Makefile [hunk 3]
```diff
@@ -19,73 +22,110 @@ export KUBEBUILDER_ASSETS
...
+.PHONY: manifests
+manifests: ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+
+.PHONY: generate
+generate: ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+
+.PHONY: fmt
+fmt: ## Run g\o fmt against code.
```
Comment (llamerada-jp) [inline Makefile:62]:
```suggestion
fmt: ## Run go fmt against code.
```

**Review by llamerada-jp — CHANGES_REQUESTED** [review]

File: config/rbac/role.yaml (hunk 2 above)
Comment (bells17) [reply config/rbac/role.yaml:7]: Sorry, I'll fix it. 
> By kubebuilder?

Yes.

Comment (bells17) [reply config/rbac/role.yaml]: fixed it.
https://github.com/topolvm/pvc-autoresizer/pull/41/commits/250ebe678a3ebe4842c90c20da6f0f3d97abaa86

Comment (bells17) [reply config/rbac/role.yaml]: Note: https://github.com/kubernetes-sigs/controller-tools/issues/513#issuecomment-722219300

Comment (bells17) [reply config/rbac/role.yaml]: Scaffolding code is here: https://github.com/kubernetes-sigs/kubebuilder/blob/v3.1.0/pkg/plugins/golang/v2/scaffolds/internal/templates/config/rbac/role_binding.go#L32-L55

**Comment by bells17** [comment]
@toshipp I fixed the role name. Would you review again?

File: Makefile (hunk 3 above)
Comment (bells17) [reply Makefile]: Thank you. I applied this suggested change.

File: Makefile [hunk 4]
```diff
@@ -1,9 +1,12 @@
 # Makefile for pvc-autoresizer
 
-K8S_VERSION = 1.18.9
-KUBEBUILDER_VERSION = 2.3.1
-CTRLTOOLS_VERSION = 0.5.0
-KUSTOMIZE_VERSION = 3.7.0
+ENVTEST_K8S_VERSION = 1.19.2
+KUBEBUILDER_VERSION = 3.1.0
+CTRLTOOLS_VERSION = 0.4.1
```
Comment (toshipp) [inline Makefile:5]: we use v0.5.0 instead of kubebuilder default version. Would you set this to previous version?

**Review by toshipp — CHANGES_REQUESTED** [review]

File: Makefile (hunk 4 above)
Comment (bells17) [reply Makefile:5]: The diff of controller-tools from v0.4.1 to v0.5.0:
https://github.com/kubernetes-sigs/controller-tools/compare/v0.4.1...v0.5.0

- https://github.com/kubernetes-sigs/controller-tools/pull/522
- https://github.com/kubernetes-sigs/controller-tools/pull/528
- https://github.com/kubernetes-sigs/controller-tools/pull/534
- https://github.com/kubernetes-sigs/controller-tools/pull/538
- https://github.com/kubernetes-sigs/controller-tools/pull/511

Comment (bells17) [reply Makefile:5]: The diff of controller-tools from v0.5.0 to v0.6.0:
https://github.com/kubernetes-sigs/controller-tools/compare/v0.5.0...v0.6.0

- https://github.com/kubernetes-sigs/controller-tools/pull/557
- https://github.com/kubernetes-sigs/controller-tools/pull/563
- https://github.com/kubernetes-sigs/controller-tools/pull/565
- https://github.com/kubernetes-sigs/controller-tools/pull/541
- https://github.com/kubernetes-sigs/controller-tools/pull/564
- https://github.com/kubernetes-sigs/controller-tools/pull/576

Comment (bells17) [reply Makefile]: Sorry @toshipp, If fixed version.
https://github.com/topolvm/pvc-autoresizer/pull/41/commits/bedef6f2642954f727b17fe1397bf63c30aee164

BTW, We can upgrade controller-tools to v0.6.0 too.
Should we upgrade controller-tools to v0.6.0?
Details of upgrade is here: https://github.com/topolvm/pvc-autoresizer/pull/41#discussion_r654900539

I think this upgrade doesn't have any effect on pvc-autoresizer probably.

Comment (toshipp) [reply Makefile]: For TopoLVM, we update tools as well as libraries, but for pvc-autoresizer, we have no specific update policies.
I also think the update is safe, I welcome to accept the upate.

Comment (bells17) [reply Makefile]: Thank you. I updated controller-tools to v0.6.0.
https://github.com/topolvm/pvc-autoresizer/pull/41/commits/17a6935b02036fecd1f4bf1cabc0748ceecf89ca

**Review by toshipp — APPROVED** [review]
LGTM

**Review by llamerada-jp — APPROVED** [review]
LGTM
